src/Fields/kernel_computed_field.jl

```diff
@@ -0,0 +1,103 @@
+using Oceananigans: AbstractModel
+using Oceananigans.Grids
+using Oceananigans.Utils: tupleit
+
+struct KernelComputedField{X, Y, Z, S, A, G, K, F, P} <: AbstractField{X, Y, Z, A, G}
+                  data :: A
+                  grid :: G
+                kernel :: K
+    field_dependencies :: F
+            parameters :: P
+                status :: S
+
+    """
+        KernelComputedField(loc, kernel, grid)
+
+    Example
+    =======
+
+    ```julia
+    using KernelAbstractions: @index, @kernel
+    using Oceananigans.Fields: AveragedField, KernelComputedField, compute!
+    using Oceananigans.Grids: Cell, Face
+    
+    @inline ψ²(i, j, k, grid, ψ, Ψ) = @inbounds (ψ[i, j, k] - Ψ[i, j, k])^2
+    @kernel function compute_var!(var, grid, ϕ, Φ)
```

Comment:
I know it's just an example but would `compute_perturbation!` be a better name?

Reply:
I'm open to a change of names, but my idea was that it computes the variance, hence `compute_var!`. (Although in hindsight I should have just named it `compute_variance!` and be more explicit.

Reply:
Ah haha sorry I thought it was short for `compute_variable!`. Indeed `compute_variance!` would be more clearer.

Reply:
Done!